src/physicsCalculator/GravityCalculator.cpp

```diff
@@ -0,0 +1,40 @@
+#include "GravityCalculator.h"
+
+#include <spdlog/spdlog.h>
+
+namespace physicsCalculator {
+    GravityCalculator::GravityCalculator(const Environment& new_env)
+        : Calculator(new_env) {
+        // Initialize the forces
+        calculateF();
+    }
+
+    GravityCalculator::GravityCalculator(const Environment& new_env, const std::vector<Particle>& particles, const bool init_forces) {
+        spdlog::warn("Called a GravityCalculator constructor which should only be used for testing.");
+        env = new_env;
+        container = ParticleContainer(particles);
+
+        // Initialize the forces
+        if (init_forces) {
+            calculateF();
+        }
+    }
+
+    GravityCalculator::~GravityCalculator() = default;
+
+    void GravityCalculator::calculateF() {
+        for (auto i = container.begin(); i < container.end(); i++) {
+            for (auto j = i + 1; j < container.end(); j++) {
```

Comment:
you could consider creating an iterator that returns pair of particles, so it there are 4 particles lets say, 0, 1, 2, 3, the functions:
begin: will return (0,1)
update: should update it to (0,2) -> (0,3) -> (1,2) ... so on
and end: will return (2,3),

and then this can be reused for all forces.